packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -274,14 +277,27 @@ export default class AddFragment extends BaseDialog<AddFragmentModel> {
             targetAggregation: targetAggregation ?? 'content'
         };
 
-        const command = await this.commandExecutor.getCommand(
+        const command = await this.commandExecutor.getCommand<AddFragmentChangeContentType>(
```

Comment:
I think passing an interface here for the command is not right especially AddFragmentChangeContentType. I would expect when calling getCommand to get the interface for the command but I see here some `AddFragmentChangeContentType` interface which is not relatable to the command. Please take a look at the comment below for more clarification.

Reply:
Command is directly related to a change that it generates so I think it makes sense to specify what kind of change will be created during the creation of command. You can see the parameters passed to the function an you can make sure that the passed type matches the parameters. 

I think adding generics to the command functions instead is less optimal, because it is easier to make mistakes that way. When calling those functions you don't actually know how the command was created and what  the command is.

It would be better to have the whole change type as a paramete, also try to match parameters and have different command types, but that would require a lot more changes and I don't think it makes sense to add additional changes here. We have a dedicated issue for fixing/improving types for changes and commands #2220.